AmplifyPlugins/API/AWSAPICategoryPlugin/Reachability/NetworkReachabilityNotifier.swift

```diff
@@ -15,14 +15,14 @@ class NetworkReachabilityNotifier {
     private var reachability: NetworkReachabilityProviding?
```

Comment:
Would it add too much complexity to define the notifier behavior as a protocol without availability restrictions, and then expose a `publisher` property `@available(iOS 13.0, *)`? That way it is easy to add some other notification API without availability restrictions when the time comes to support < 13.

Reply:
I agree that ideally the reachability interface wouldn't require iOS 13.0.  I think it might be a bit difficult to get in right away due to the us returning an AnyPublisher.  How about I look into this afterwards?  It doesn't seem like a high priority issue considering that this is currently only needed for the sync engine which also depends on iOS 13.

Reply:
I tried doing this, but wasn't able to figure out how to work around all the compiler complaints :(. punting.